### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by golluroSICKAG**
# Version 1.0.0
- Initial version

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

### CODE REVIEW

File: CSK_Module_Commands/project.mf.xml [hunk 1]
```diff
@@ -47,16 +51,74 @@ ToDo +
...
+                <event name="OnNewStatusParameterType">
+                    <desc>Event to notify currently selected parameter values to use for function call.</desc>
+                    <param desc="Type of parameter." multiplicity="1" name="parameterType" type="string"/>
+                </event>
+                <event name="OnNewStatusParameterAmount">
+                    <desc>Notify amount of parameters to use for function call.</desc>
+                    <param desc="Amount of parameters." multiplicity="1" name="amount" type="int"/>
+                </event>
+                <event name="OnNewLog">
+                    <desc>NJotify log messages.</desc>
```
Comment (tarnamaSICKAG) [inline CSK_Module_Commands/project.mf.xml]: Notify*

File: CSK_Module_Commands/project.mf.xml [hunk 2]
```diff
@@ -65,10 +127,87 @@ ToDo +
...
+                </function>
+                <function name="removeCommandViaUI">
+                    <desc>Function to remove command within command list.</desc>
+                </function>
+                <function name="runCommands">
+                    <desc>Function to run all commands of currently active list.</desc>
+                </function>
+                <function name="sleep">
+                    <desc>Function to call 'Script.sleep'.</desc>
+                    <param desc="Time to sleep." multiplicity="1" name="time" type="int"/>
```
Comment (tarnamaSICKAG) [inline CSK_Module_Commands/project.mf.xml]: maybe add unit: "in ms"

File: CSK_Module_Commands/project.mf.xml [hunk 3]
```diff
@@ -6,15 +6,19 @@
         </crown>
         <crown name="CSK_Commands">
             <trait>released</trait>
-            <desc>Please fill in information regarding: What is the idea of this module and its features? +
-How to use this module in general.... like: + 
+            <desc>Module to dynamically call served functions or notify served events. +
```
Comment (dzekomiSICKAG) [inline CSK_Module_Commands/project.mf.xml]: After reading this I am not sure what does this do exactly. The question is why use this and when is it going to make my life easier, what pain does it kill?

File: CSK_Module_Commands/project.mf.xml [hunk 4]
```diff
@@ -6,15 +6,19 @@
...
-**2) Something else...** +
-</desc>
+** 1) Setup the command** +
+Select the type of command via 'setMode'. +
+Set the name of the function to call or event to notify ('setFunctionName' or 'setEventName'). +
+Optionally you can add parameters to the function / event. To do so, add parameters via 'setParameterAmount', selecte a parameter via 'setSelectedParameter', set the type of the parameter via 'setParameterType' and set the value via 'setParameterValue'. +
+{empty} + 
+** 2) List of commands** +
+It is possible to set a list of commands to be called / notified one after another. +
+You can add a predefined command via 'addCommandViaUI' to that list and start the processing via 'runCommands'.</desc>
```
Comment (dzekomiSICKAG) [inline CSK_Module_Commands/project.mf.xml]: Also after reading it I don't know if this is the typical "observer" type tool, but with some GUI. I know I can call functions and raise Events. Maybe a bit more marketing style language here but still keep technical details

File: README.md [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
 
 ## Information
 
-Tested on:
+Tested on:  
 |Device|Firmware|Module version
 |--|--|--|
-|e.g. SICK AppEngine|V1.5.0|V0.1.0|
+|SICK AppEngine|V1.7.0|V1.0.0|
+|SIM1012|V2.4.2|V1.0.0|
```
Comment (dzekomiSICKAG) [inline README.md:17]: SICK AppEngine 2.0.0 TDC-E

**Comment by kogliinSICKAG** [comment]
I also used the Commands module while testing the FlowConfig module and found a runtime error:
When you use the default settings with CSK_Commands.print which is without parameters, it runs into these two error lines:
[09:55:30.010: WARNING: API.CSK_Commands: App 'CSK_Module_Commands': Pos 'System/Commands/Commands_Model.lua:119'] Function call 'CSK_Commands.print' blocked due to wrong parameters: Number of values should be 1 but is 0
[09:55:30.011: WARNING: AppEngine.SharedLogger.ModuleLogger: App 'CSK_Module_Commands': Pos 'System/Commands/Commands_Model.lua:122'] CSK_Commands: No success to call 'CSK_Commands.print'

**Comment by kogliinSICKAG** [comment]
Just checked the docu for curiosity on how to use the commands but unfortunately, the docu still contains the short description of the template:
image
Is this maybe still the docu of the unmerged module? I followed the link to the docu within your repository but I think that was not the right way.
I found the description within the repository: One suggestion: For the list of the commands you tell that the lsit can be executed with "runCommands". Is this the same for executing the single command? I think not as the GUI also has a different button for that. Thus, it would be good to mention that for the single command too, maybe like run single command via "executeCommand"?

**Comment by kogliinSICKAG** [comment]
If I look into the module "FlowConfig", I cannot find the module "Commands" as a block there. Nevertheless, the module "Commands" has the checkbox for "flow config priority". Why does it have that?

**Review by dzekomiSICKAG — APPROVED** [review]

**Comment by golluroSICKAG** [comment]
Thx for the review!
All mentioned issues should be fixed.
@kogliinSICKAG : Regarding the html documentation: Yes, a fixed link is needed, so the update is only visibile on your local file...

**Pull request merged**